python-package/lightgbm/engine.py

```diff
@@ -131,12 +131,16 @@ def train(params, train_data, num_boost_round=100,
         predictor = init_model
     else:
         predictor = None
+    init_iteration = predictor.init_iteration if predictor else 0
```

Comment:
I think init_iteration should be inside of Booster. Don't expose it. 
Early_stopping still return the relative iteration. 
And use init_iteration+passed_iter as final iteration to predict.